woocommerce/woocommerce-android
# Networking support gift cards analytics

Closes: #10969 
Closes: #10971

### Why
This PR connects the Gift Cards stats UI with the data obtained from the API request.

### Description
This PR adds the fetch gift cards stats logic starting from the gift card's rest client up to the view model. For all the actors in between (StatsRepository, AnalyticsRepository, UpdateAnalyticsHubStats), I've added unit tests to cover the major scenarios. 

### Testing instructions
** Need to have the Gift Card extension installed and active

1. Open the app
2. Tap on See all store analytics
3. Check that the gift cards card is displayed depending on the current Analytic Hub configuration
4. Check that the stats values match the ones on `wp-admin`

### Images/gif
<img width="300" src="https://github.com/woocommerce/woocommerce-android/assets/18119390/e960acc5-7d5b-4865-ae28-df469d34f474" />
